### PR TITLE
feat: Propagate the plugin registry and dashboard urls to the containers in the initial devworkspace templates

### PR DESCRIPTION
**Pull request by JPinkney**
…ers in the initial devworkspace templates

Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### What does this PR do?
This PR propagates the plugin registry and dashboard urls to every container in the initial devworkspacetemplate

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/19933

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
What will happen for theia in DWO without Che ? We'll have to give some env for the plugin registry url ?

**Comment by sleshchenko** [comment]
> What will happen for theia in DWO without Che ? We'll have to give some env for the plugin registry url ?

We're not supposed to run Theia in DWO without Che.
If you mean our samples, probably we could manually update some of samples for testing particular Theia functionality, but if we want Theia be able to run without Che, dashboard and plugin registry URLs must be optional.

File: src/services/workspace-client/devWorkspaceClient.ts [hunk 1]
```diff
@@ -48,6 +48,8 @@ export class DevWorkspaceClient extends WorkspaceClient {
   private initializing: Promise<void>;
   private lastDevWorkspaceLog: Map<string, string>;
   private devWorkspacesIds: string[];
+  private pluginRegistryUrlName: string;
+  private dashboardUrlName: string;
```
Comment (sleshchenko) [inline src/services/workspace-client/devWorkspaceClient.ts:52]:
```suggestion
  private dashboardUrlEnvName: string;
```
and the same above

**Review by sleshchenko — APPROVED** [review]
I haven't checked yet but LGTM in general. Just a few minor suggestions

**Review by akurinnoy — APPROVED** [review]

**Review by olexii4 — APPROVED** [review]

**Pull request merged**